src/query/ast/src/parser/expr.rs

```diff
@@ -1387,15 +1387,15 @@ pub fn type_name(i: Input) -> IResult<TypeName> {
             | #ty_bitmap
             | #ty_tuple : "TUPLE(<type>, ...)"
             | #ty_named_tuple : "TUPLE(<name> <type>, ...)"
-            ) ~ NULL? : "type name"
+            ) ~ (NOT|NULL)? : "type name"
             },
             rule! {
             ( #ty_date
             | #ty_datetime
             | #ty_string
             | #ty_variant
             | #ty_nullable
-            ) ~ NULL? : "type name" },
+            ) ~(NOT|NULL)? : "type name" },
```

Comment:
this is incorrect, `not null` will cause a parse error, we need to support both parses `not null` and `null`.

Reply:
sure